src/test/java/com/google/sps/CommentAnalysisTest.java

```diff
@@ -14,70 +14,153 @@
+import com.google.api.services.youtube.model.CommentThread;
 import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.api.services.youtube.model.CommentThreadSnippet;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
 import com.google.sps.servlets.utils.CommentAnalysis;
+import com.google.sps.servlets.utils.Range;
 import com.google.sps.servlets.utils.Statistics;
-import java.io.IOException;
-import java.security.GeneralSecurityException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
 
-/** This is a JUnit test for sentiment analysis */
+/** This is a JUnit test for sentiment mockedAnalysis */
 @RunWith(JUnit4.class)
 public class CommentAnalysisTest {
-  private CommentThreadListResponse youtuberesponse;
-  private CommentAnalysis analysis;
-  private static final String APPLICATION_NAME = "testComment";
-  private static final String DEVELOPER_KEY = "API_KEY";
-  private static final String PLAINTEXT = "plainText";
-  private static final String SNIPPET = "snippet";
-  private static final String TEST_VIDEO_ID = "E_wKLOq-30M";
-  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final float TEST_SCORE = 0.23f;
+  private static Comment testTopComment = new Comment();
+  private static CommentSnippet topCommentSnippet = new CommentSnippet();
+  private static CommentThread testCommentThread = new CommentThread();
+  private static List<CommentThread> testCommentThreadList =
+      new ArrayList<>(Arrays.asList(testCommentThread, testCommentThread));
+  private static CommentThreadListResponse youtubeResponse = new CommentThreadListResponse();
+  private static CommentThreadSnippet testThreadSnippet = new CommentThreadSnippet();
+  private static LanguageServiceClient mockedlanguageService =
+      mock(LanguageServiceClient.class, Mockito.RETURNS_DEEP_STUBS);
+  private static CommentAnalysis commentAnalysis = new CommentAnalysis(mockedlanguageService);
 
-  /**
-   * Build and return an authorized API client service.
-   *
-   * @return an authorized API client service
-   * @throws GeneralSecurityException, IOException
-   */
-  public static YouTube getService() throws GeneralSecurityException, IOException {
-    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-    return new YouTube.Builder(httpTransport, JSON_FACTORY, null)
-        .setApplicationName(APPLICATION_NAME)
-        .build();
+  @Before
+  public void setUp() {
+    topCommentSnippet.setTextDisplay("Test Message");
+    testTopComment.setSnippet(topCommentSnippet);
```

Comment:
Optional change:

Since most of these `set*` methods return themselves, you can make these definitions more compact by moving them to a single line.

I.e. (formatting/syntax might be wrong) something like:
`
Comment comment = new Comment().setSnippet((new CommentSnippet()).setTextDisplay("Test Message"));`